src/test/java/org/mule/service/scheduler/internal/RunnableFutureDecoratorTestCase.java

```diff
@@ -50,47 +43,6 @@ public void after() throws Exception {
     super.after();
   }
 
-  @Test
-  public void threadLocalValuesNotLeaked() throws InterruptedException, ExecutionException {
-    final ThreadLocalHolder tlHolder = new ThreadLocalHolder();
-
-    Object valueToStore = new Object();
-    PhantomReference ref = new PhantomReference<>(valueToStore, new ReferenceQueue<>());
-    submitStore(tlHolder, valueToStore);
-
-    valueToStore = null;
-
-    new PollingProber(GC_POLLING_TIMEOUT, DEFAULT_POLLING_INTERVAL)
-        .check(new JUnitLambdaProbe(() -> {
-          System.gc();
-          assertThat(ref.isEnqueued(), is(true));
-          return true;
-        }, "A hard reference is being mantained to the value stored in a thread local."));
-
-  }
-
-  private void submitStore(final ThreadLocalHolder tlHolder, Object valueToStore)
-      throws InterruptedException, ExecutionException {
-    scheduler.submit(() -> {
-      tlHolder.set(valueToStore);
-    }).get();
-  }
-
-  public static final class ThreadLocalHolder {
-
-    private final ThreadLocal threadLocal = new ThreadLocal();
-
-    public void set(Object value) {
-      threadLocal.set(value);
-    }
-  }
-
-  @Test
-  public void mdcValuesNotSpreadAmongTasks() throws InterruptedException, ExecutionException {
-    submitMdcPut();
-    submitMdcGet();
-  }
-
```

Comment:
All these tests have been moved to the SchedulerThreadPool cases because it is no longer a responsibility of RunnableFutureDecorator.